woocommerce/woocommerce-android
# Connecting shipping rates

Part of: #12998

### Description
This PR connects shipping rates with the mock data returned from the `GetShippingRates` use case. Additionally, it adds logic for changing the sort order and refreshing the shipping rates.

Changes included here:
- Display shipping rates from the `GetShippingRates` use case.
- Sort shipping rates using the sort order selected
- Refresh shipping rates (used in the error screen)
- Added a loading state to indicate that the app is working while we are fetching shipping rates

### Testing information
TC1
- Open the app
- Navigate to orders
- Tap on an order eligible for creating a shipping label
- Tap on create a shipping label
- Check that the shipping rates are displayed
- Change the shipping sort order
- Check that shipping rates are refreshed

TC2
First apply this patch to force the get shipping labels use case to fail

<details>
  <summary>Patch</summary>

```
Index: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/GetShippingRates.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/GetShippingRates.kt b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/GetShippingRates.kt
--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/GetShippingRates.kt	(revision 063540b77d705976f703f59726e538865681c073)
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/GetShippingRates.kt	(date 1733926052842)
@@ -5,6 +5,7 @@
 import kotlinx.coroutines.delay
 import javax.inject.Inject
 import kotlin.random.Random
+
 @Suppress("MagicNumber")
 class GetShippingRates @Inject constructor() {
     private val cheapestComparator = Comparator<ShippingRateUI> { r1, r2 ->
@@ -62,13 +63,14 @@
             )
         }
 
-        return Result.success(
+        /*return Result.success(
             carriers.associateWith {
                 generateRates(
                     it.name,
                     Random(0).nextInt(from = 3, until = 10)
                 ).sortedWith(comparator)
             }
-        )
+        )*/
+        return Result.failure(Exception("Random exception"))
     }
 }

```

</details>

- Open the app
- Navigate to orders
- Tap on an order eligible for creating a shipping label
- Tap on create a shipping label
- Check that an error is displayed
- Tap on refresh
- Check that shipping labels refresh (displaying the error again)

### The tests that have been performed
- Testing that the UI works as expected

### Images/gif

https://github.com/user-attachments/assets/c867a224-24ee-4e8e-a15f-1f37de539a77




## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:


<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md --
